044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,342 @@
+
+To make the throttle factor logic more resilient in the face of transient failures we want to continue
+using a previously calculated valid throttle factor for some duration, rather than immediately using the 
+throttle factor fallback after a single failure.
+
+So we will enable the user to specify how long a valid throttle factor can be applied for with
+`client.quota.callback.static.throttle.factor.validity-duration`. Where the value is an ISO8601 duration.
+Default value "PT5M", 5 minutes.
+
+For example if validity duration is 2 minutes then we could expect this behaviour:
+
+1. throttle factor recalculation at minute 0 succeeded, calculated throttle factor 0.6 and use it
+2. throttle factor recalculation at minute 1 failed, continue using throttle factor 0.6
+3. throttle factor recalculation at minute 2 failed, continue using throttle factor 0.6
+4. throttle factor recalculation at minute 3 failed, use throttle factor fallback of 1.0
+5. throttle factor recalculation at minute 4 succeeded, calculated throttle factor 0.6 and use it
+
+## Configuration Summary
+
+|                                                                | type   | default | valid values      |                                                                                                                                  |
+|----------------------------------------------------------------|--------|---------|-------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| client.quota.callback.static.storage.volume.source             | string | local   | local, cluster    | set to cluster to obtain volume descriptions from cluster, see [VolumeSource](#volume-source)                                    |
+| client.quota.callback.static.storage.check-interval            | long   | 0       | (0, ...)          | 0 means disabled, otherwise this is the milliseconds between polling to describe the cluster (this is an existing configuration) |
+| client.quota.callback.static.kafka.admin.bootstrap.servers     | string | null    |                   | required if volume source is cluster, bootstrap.servers for [admin client](#admin-client-configuration) used to get cluster data |
+| client.quota.callback.static.kafka.admin.*                     | ?      |         |                   | optionally users can configure arbitrary properties of the [admin client config](#admin-client-configuration)                    |
+| client.quota.callback.static.throttle.factor.fallback          | double | 1.0     | (0.0, 1.0)        | sets the [Throttle Factor Fallback](#throttle-factor-fallback)                                                                   |
+| client.quota.callback.static.throttle.factor.validity-duration | string | PT5M    | ISO8601 durations | sets the [Throttle Factor Validity Duration](#throttle-factor-validity-duration)                                                 |
```

Comment:
Do we have other configs using dashes as separators?

Reply:
There was a pre-existing property `check-interval` using dashes, but we've pushed up a change to use dots to align with kafka.